bobtask/verify.go

```diff
@@ -16,9 +24,36 @@ func (t *Task) VerifyAfter() error {
 }
 
 func (t *Task) verifyBefore() (err error) {
+	if t.target != nil {
+		for _, path := range t.target.FilesystemEntriesRawPlain() {
+			if !isValidFilesystemTarget(path) {
+				return usererror.Wrap(fmt.Errorf("invalid target `%s` for task `%s`", path, t.name))
+			}
+		}
+	}
+
 	return nil
 }
 
+// isValidFilesystemTarget checks if target is a valid path
+// paths which resolve to `.` or containing `..` are considered invalid
+func isValidFilesystemTarget(path string) bool {
+	// check for end with one dot
+	if path == "." {
+		return false
+	}
+
+	if strings.Contains(path, "..") {
```

Comment:
Might not be common, but something like `file..extension` should be valid.

Reply:
Thx for mentioning it. No leveraging `filepath.Clean()` which should handle those cases correctly.